packages/connect/src/device/DeviceList.ts

```diff
@@ -59,99 +47,293 @@ export interface DeviceList {
 
-        const { env } = DataManager.settings;
-        let { transports: transportSettings } = DataManager.settings;
-
-        const transports: Transport[] = [];
-
-        // fallback of "the last instance".
-        // since web index (that otherwise set fallbacks) is released into npm, we can not rely
+        // we fill in `transports` with a reasonable fallback in src/index.
+        // since web index is released into npm, we can not rely
         // on that that transports will be always set here. We need to provide a 'fallback of the last resort'
-        if (!transportSettings?.length) {
-            transportSettings = ['BridgeTransport', 'WebUsbTransport'];
-        }
-
-        if (env === 'react-native' && typeof ReactNativeUsbPlugin !== 'undefined') {
-            transports.push(ReactNativeUsbPlugin());
-        } else if (transportSettings?.includes('BridgeTransport')) {
-            const bridgeLatestVersion = getBridgeInfo().version.join('.');
-            const bridge = new BridgeV2(undefined, undefined);
-            bridge.setBridgeLatestVersion(bridgeLatestVersion);
-
-            this.fetchController = getAbortController();
-            const { signal } = this.fetchController;
-            // @ts-expect-error TODO: https://github.com/trezor/trezor-suite/issues/5332
-            const fetchWithSignal = (args, options = {}) => fetch(args, { ...options, signal });
-
-            // detection of environment browser/node
-            // todo: code should not be detecting environment itself. imho it should be built with this information passed from build process maybe?
-            const isNode =
-                !!process?.release?.name && process.release.name.search(/node|io\.js/) !== -1;
-            BridgeV2.setFetch(fetchWithSignal, isNode);
-            // @ts-expect-error TODO: https://github.com/trezor/trezor-suite/issues/5332
-            transports.push(bridge);
+        if (!transports?.length) {
+            transports = ['BridgeTransport', 'WebUsbTransport'];
         }
 
-        if (transportSettings?.includes('WebUsbTransport') && typeof WebUsbPlugin !== 'undefined') {
-            transports.push(WebUsbPlugin());
-        }
+        // mapping of provided transports[] to @trezor/transport classes
+        transports.forEach(transportType => {
+            if (typeof transportType === 'string') {
+                switch (transportType) {
+                    case 'NodeUsbTransport':
+                        this.transports.push(new NodeUsbTransport({ messages: this.messages }));
+                        break;
+                    case 'WebUsbTransport':
+                        // todo: any idea how to configure typescript here?
+                        this.transports.push(
+                            // @ts-expect-error
+                            new WebUsbTransport({
+                                messages: this.messages,
+                            }),
+                        );
```

Comment:
I ended up doing it like this c695d43f1d33c019c7a6e80dd5cf7ea35862de2f changing packages/transport/src/transports/webusb.ts type definition in that way that it satisfies abstract transport types. it required few expect-errors but these are ok since this file (packages/transport/src/transports/webusb.ts) is meant to by replaced by webusb.browser anyway